### PR TITLE
Regenerate reference docs from in-repo sources

### LINKED ISSUE #37: Auto-synced reference docs don't match checked-in code, thus sync PRs have spurious changes
It seems the current content for the reference docs (scraped from readme?) doesn't match those generated by the auto-sync PR, which means the doc syncing PRs don't fit into that smoothly: e.g. #32 has a massive diff stat, despite being a no-op update, theoretically (readme should be showing version 2.18.1 already).

I imagine these differences may result in mis-rendered docs, if they were merged.

Notable differences in reference/generated docs:

| # | PR behaviour | main behaviour | PR link | `main` link |
|---|---|---|---|---|
|1| only a few subsystems (e.g. no `add-trailing-comma`) | lots of subsystems | [PR](https://github.com/pantsbuild/pantsbuild.org/tree/f7004fbcc5e386abf998603de17a839546cef4c2/versioned_docs/version-2.18.x/reference/subsystems) | [main](https://github.com/pantsbuild/pantsbuild.org/tree/6185bd4d59545d3f5f0e473939eaff743e3f0054/versioned_docs/version-2.18.x/reference/subsystems) |
|2| spacing differences | - | [PR diff of `anonymous-telemetry.mdx`](https://github.com/pantsbuild/pantsbuild.org/pull/32/files#diff-8539553f255b37b9d6ae0fced75385e6ca4166f0f4cd05e1e83469ee52c862d5) | (see diff) |
|3| escaping etc. differences | - | [PR diff of `target.mdx`](https://github.com/pantsbuild/pantsbuild.org/pull/32/files#diff-f167a31ae9026b13c3744ce353e50d853fa99710381e30a449773758e78ca7b0) | (see diff) |
|4| no related subsystems in goals | - | [PR diff of `check.mdx`](https://github.com/pantsbuild/pantsbuild.org/pull/32/files#diff-0ef5502578202a722dd7a62dd567c502f61bc3c2658c86ba3a3031b05d46d849) | (see diff) |
|5| `default_repr` differences | - | [PR diff of `global-options.mdx`](https://github.com/pantsbuild/pantsbuild.org/pull/32/files#diff-53d8f10c04de6d0452ceda55c6708e8cf2e297b1f4b5f508f48b879f5454fe8c) | (see diff) |

Potentially others: would be easiest to identify once we solve more of those to reduce the number of diffs to filter through.

(See #34 for differences in the prose docs)

### PR DESCRIPTION
**Pull request by huonw**
This runs `for d in docs versioned_docs/*; do npm run generate-reference $d; done`, `npm run format-all` to regenerate all docs off the `help-all.json` files that are now checked in (#43).

Fixes #37, hopefully.

### CODE REVIEW

File: versioned_docs/version-2.0.x/reference/global-options.mdx [hunk 1]
```diff
@@ -559,7 +563,7 @@ Directory to use for named global caches for tools and processes with trusted, c
 <Option
   cli_repr={`--local-execution-root-dir=<str>`}
   env_repr='PANTS_LOCAL_EXECUTION_ROOT_DIR'
-  default_repr={`/tmp`}
+  default_repr={`<buildroot>`}
```
Comment (huonw) [inline versioned_docs/version-2.0.x/reference/global-options.mdx:566]: Is this change correct?

File: versioned_docs/version-2.0.x/reference/goals/awslambda.mdx [hunk 2]
```diff
@@ -53,19 +53,5 @@ None
...
-- [lambdex](../subsystems/lambdex.mdx)
-- [pex](../subsystems/pex.mdx)
-- [protoc](../subsystems/protoc.mdx)
-- [python-infer](../subsystems/python-infer.mdx)
-- [python-native-code](../subsystems/python-native-code.mdx)
-- [python-protobuf](../subsystems/python-protobuf.mdx)
-- [python-repos](../subsystems/python-repos.mdx)
-- [python-setup](../subsystems/python-setup.mdx)
-- [source](../subsystems/source.mdx)
-- [subprocess-environment](../subsystems/subprocess-environment.mdx)
```
Comment (huonw) [inline versioned_docs/version-2.0.x/reference/goals/awslambda.mdx:69]: This is lost?

File: versioned_docs/version-2.0.x/reference/goals/awslambda.mdx [hunk 3]
```diff
@@ -13,14 +13,14 @@ pants awslambda [args]
 
 Deprecated in favor of the `package` goal.
 
-Backend: <span className="color--primary">`pants.core`</span>
-
-Config section: <span className="color--primary">`[awslambda]`</span>
+Backend: <span className="color--primary"></span>
+Config section: <span className="color--primary">[awslambda]</span>
```
Comment (huonw) [inline versioned_docs/version-2.0.x/reference/goals/awslambda.mdx:17]: I wonder if this spacing change will cause problems?

File: versioned_docs/version-2.0.x/reference/targets/archive.mdx [hunk 4]
```diff
@@ -18,61 +18,62 @@ Backend: <span className="color--primary">``</span>
...
 Arbitrary strings that you can use to describe a target. For example, you may tag some test targets with &#x27;integration_test&#x27; so that you could run `./pants --tags='integration_test' test ::` to only run on targets with that tag.
 
 </Field>
-
 ## `description`
 
 <Field
-    type_repr={`field.type_hint`}
+    type_repr={`str | None`}
+    default_repr={`None`}
```
Comment (huonw) [inline versioned_docs/version-2.0.x/reference/targets/archive.mdx:32]: A lot of targets now pass this ``default_repr={`None`}``. Is that a problem?

File: versioned_docs/version-2.0.x/reference/targets/archive.mdx [hunk 5]
```diff
@@ -18,61 +18,62 @@ Backend: <span className="color--primary">``</span>
...
 >
 
 Where the built asset should be located. If undefined, this will use the path to the the BUILD, followed by the target name. For example, `src/python/project:app` would be `src.python.project/app.ext`. When running `./pants package`, this path will be prefixed by `--distdir` (e.g. `dist/`). Warning: setting this value risks naming collisions with other package targets you may have.
 
 </Field>
-
```
Comment (huonw) [inline versioned_docs/version-2.0.x/reference/targets/archive.mdx:47]: I wonder if this spacing change is a problem?

File: versioned_docs/version-2.0.x/reference/targets/relocated_files.mdx [hunk 6]
```diff
@@ -17,14 +17,14 @@ For example, you can relocate `src/resources/project1/data.json` to instead be
...
-
 ```
 # Results in `data.json`.
 relocated_files(
     file_targets=["src/resources/project1:target"],
     src="src/resources/project1",
     dest="",
 )
+
 ```
```
Comment (huonw) [inline versioned_docs/version-2.0.x/reference/targets/relocated_files.mdx:28]: It looks like the spacing has gone a bit weird here.

**Review by huonw — COMMENTED** [review]
It looks like there's a few places where the generation script potentially has the wrong behaviour.

File: versioned_docs/version-2.0.x/reference/goals/awslambda.mdx (hunk 2 above)
Comment (thejcannon) [reply versioned_docs/version-2.0.x/reference/goals/awslambda.mdx:69]: I think so

File: versioned_docs/version-2.0.x/reference/global-options.mdx (hunk 1 above)
Comment (thejcannon) [reply versioned_docs/version-2.0.x/reference/global-options.mdx:566]: That one should be `<tmpdir>`: https://www.pantsbuild.org/docs/reference-global#local_execution_root_dir

File: versioned_docs/version-2.0.x/reference/targets/archive.mdx (hunk 5 above)
Comment (thejcannon) [reply versioned_docs/version-2.0.x/reference/targets/archive.mdx:47]: Possibly

File: versioned_docs/version-2.0.x/reference/targets/archive.mdx (hunk 4 above)
Comment (thejcannon) [reply versioned_docs/version-2.0.x/reference/targets/archive.mdx:32]: That was a bug in the old templates, it seems. See https://www.pantsbuild.org/docs/reference-archive

File: versioned_docs/version-2.0.x/reference/goals/awslambda.mdx (hunk 2 above)
Comment (thejcannon) [reply versioned_docs/version-2.0.x/reference/goals/awslambda.mdx:69]: Fixed in https://github.com/pantsbuild/pantsbuild.org/pull/53

File: versioned_docs/version-2.0.x/reference/goals/awslambda.mdx (hunk 3 above)
Comment (thejcannon) [reply versioned_docs/version-2.0.x/reference/goals/awslambda.mdx:17]: Fixed in https://github.com/pantsbuild/pantsbuild.org/pull/53

File: versioned_docs/version-2.0.x/reference/targets/archive.mdx (hunk 5 above)
Comment (thejcannon) [reply versioned_docs/version-2.0.x/reference/targets/archive.mdx:47]: Fixed in https://github.com/pantsbuild/pantsbuild.org/pull/53

File: versioned_docs/version-2.0.x/reference/targets/relocated_files.mdx (hunk 6 above)
Comment (thejcannon) [reply versioned_docs/version-2.0.x/reference/targets/relocated_files.mdx:28]: Fixed in https://github.com/pantsbuild/pantsbuild.org/pull/53

Comment (thejcannon) [reply versioned_docs/version-2.0.x/reference/targets/relocated_files.mdx:28]: Although, how did you re-generate the docs here? That red line should've been fixed by `prettier`. Did you run` prettier` after running the generation command?

File: versioned_docs/version-2.0.x/reference/global-options.mdx (hunk 1 above)
Comment (thejcannon) [reply versioned_docs/version-2.0.x/reference/global-options.mdx:566]: Ah yeah, this looks like a bug in the generations script, at some point I think `pants_workdir` changed defaults.

File: versioned_docs/version-2.0.x/reference/targets/relocated_files.mdx (hunk 6 above)
Comment (huonw) [reply versioned_docs/version-2.0.x/reference/targets/relocated_files.mdx:28]: I didn't; I didn't realise I had to :)

Comment (thejcannon) [reply versioned_docs/version-2.0.x/reference/targets/relocated_files.mdx:28]: If there was a way to have it run in the script definition (e.g. run the generation and format) id +1 that

File: versioned_docs/version-2.0.x/reference/global-options.mdx (hunk 1 above)
Comment (huonw) [reply versioned_docs/version-2.0.x/reference/global-options.mdx]: I think we can let this one go, it looks like a "problem" in existing docs before 2.13:

- 2.12: https://www.pantsbuild.org/v2.12/docs/reference-global#local_execution_root_dir
image

- 2.13: https://www.pantsbuild.org/v2.13/docs/reference-global#local_execution_root_dir
image



And indeed, the docs match, e.g. 2.20: https://github.com/pantsbuild/pantsbuild.org/blob/09d492d7962bf5535dad14ef8d28a46f14d72f45/docs/reference/global-options.mdx#L695-L696

**Comment by huonw** [comment]
(Waiting for #54 before I check over this again.)

File: versioned_docs/version-2.0.x/reference/global-options.mdx (hunk 1 above)
Comment (thejcannon) [reply versioned_docs/version-2.0.x/reference/global-options.mdx]: I split this into a dedicated issue: https://github.com/pantsbuild/pantsbuild.org/issues/60

**Comment by huonw** [comment]
I've done some _very_ quicks spot checks of the diff and of the rendering, and it seems fine.

**Review by thejcannon — APPROVED** [review]
Let's goooo!

File: versioned_docs/version-2.0.x/reference/targets/relocated_files.mdx (hunk 6 above)
Comment (huonw) [reply versioned_docs/version-2.0.x/reference/targets/relocated_files.mdx]: Done in https://github.com/pantsbuild/pantsbuild.org/pull/65#pullrequestreview-1805284823

**Pull request merged**